rpc/src/v1/helpers/block_import.rs

```diff
@@ -19,23 +19,16 @@
 use ethcore::client::BlockQueueInfo;
 use sync::SyncState;
 
-/// Check if client is during major sync or during block import and allows defining whether 'waiting for peers' should
-/// be considered a syncing state.
-pub fn is_major_importing_or_waiting(sync_state: Option<SyncState>, queue_info: BlockQueueInfo, waiting_is_syncing_state: bool) -> bool {
+/// Check if client is during major sync or during block import.
+pub fn is_major_importing(sync_state: Option<SyncState>, queue_info: BlockQueueInfo) -> bool {
```

Comment:
I think the flag was introduced quite recently, no? `WaitingForPeers` is the initial state in case we are attempting warp sync, and we don't really want to start regular sync just yet (we first wait for at least couple of warp-peers before falling back to regular sync).

Reply:
Yes it was added recently (#9112), but it's not necessary if we don't switch into `WaitingPeers` state on the beginning of each sync round, and in that case we should always consider `WaitingPeers` to be a syncing state.